### DISCUSSION TITLE
How can I stop brew upgrading all packages when I install something?

### DISCUSSION THREAD

**Discussion by johnalanwoods** [Homebrew/discussions#3280] [Everyday usage] [answered]
If I `brew install lorem` - why does brew take it upon itself to upgrade all packages, even those which are not deps?

How can this be disabled?

Thank you.

**Comment by gromgit** [comment, marked as answer]
https://docs.brew.sh/FAQ#why-does-brew-upgrade-formula-or-brew-install-formula-also-upgrade-a-bunch-of-other-stuff

And I don't recall any way to disable this behavior, since it leads over time to an inconsistent "upgrade state" that just leads to unnecessary issues being filed.

**Reply by carlocab** [reply]
`lorem` has no dependencies, so it seems like this shouldn’t be happening. We’ll need a way to reproduce it to track down the bug if there is one.

**Reply by johnalanwoods** [reply]
Thanks @carlocab ! - Lorem was placeholder text, not the Lorem package :D

**Reply by carlocab** [reply]
Right, probably not a bug then. In general `brew install foo` will upgrade all formulae connected to `foo` via the dependency graph.

It just so happens that lots of formulae are connected to each other (think anything that might depend on OpenSSL), so at a first glance it looks like a `brew install` upgrades everything.

**Comment by johnalanwoods** [comment]
Thanks both.
